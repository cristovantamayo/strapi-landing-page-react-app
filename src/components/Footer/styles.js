import styled, { css } from "styled-components";
import { theme } from "../../styles/theme";

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    a {
      color: inherit;
      text-decoration: nome;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
