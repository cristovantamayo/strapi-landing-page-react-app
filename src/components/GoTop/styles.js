import styled, { css } from "styled-components";

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4.8rem;
    height: 4.8rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease-in-out;

    &:hover {
      opacity: 1;
      background: red;
    }
  `}
`;
