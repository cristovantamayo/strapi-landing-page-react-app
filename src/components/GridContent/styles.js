import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    max-width: 60rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      max-width: 40rem;
    }
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xHuge};
  `}
`;
