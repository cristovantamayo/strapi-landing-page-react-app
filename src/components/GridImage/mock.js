export default {
  background: false,
  title: "My grid",
  description:
    "Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde",
  grid: [
    {
      altText: "Teste 1",
      srcImg:
        "https://images.unsplash.com/photo-1506748686214-e9df14d4d9d0?w=800&h=800",
    },
    {
      altText: "Teste 2",
      srcImg:
        "https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800",
    },
    {
      altText: "Teste 3",
      srcImg:
        "https://images.unsplash.com/photo-1469474968028-56623f02e42e?w=800&h=800",
    },
    {
      altText: "Teste 4",
      srcImg:
        "https://images.pexels.com/photos/460621/pexels-photo-460621.jpeg?w=800&h=800",
    },
    {
      altText: "Teste 5",
      srcImg:
        "https://images.pexels.com/photos/210186/pexels-photo-210186.jpeg?w=800&h=800",
    },
    {
      altText: "Teste 6",
      srcImg:
        "https://images.unsplash.com/photo-1496483648148-47c686dc86a8?w=800&h=800",
    },
  ],
};
