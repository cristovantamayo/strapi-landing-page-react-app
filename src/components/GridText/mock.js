export default {
  background: false,
  title: "My grid",
  description: `Atque dolorum, quibusdam
    quisquam, quas, voluptate
    quos laboriosam voluptatum,
    quisquam quisquam, quas,
    voluptate quos laboriosam
    voluptatum.`,
  grid: [
    {
      title: "Teste 1",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
    {
      title: "Teste 2",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
    {
      title: "Teste 3",
      description: `Lorem ipsum dolor sit,
      amet consectetur adipisicing elit.
      Debitis cum delectus molestias.
      Atque doloribus nobis laudantium
      esse ut, non commodi maxime
      distinctio veritatis unde,
      reprehenderit minus ad dolores
      provident maiores.`,
    },
  ],
};
