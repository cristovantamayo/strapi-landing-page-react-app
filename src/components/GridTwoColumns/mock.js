export default {
  title: "Grid Two Columns",
  text: "Lorem ipsum dolor sit amet consectetur adipisicing elit",
  srcImg: "assets/images/javascript.svg",
};
