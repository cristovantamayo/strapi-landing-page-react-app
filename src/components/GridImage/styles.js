import styled, { css } from "styled-components";
import { Title as HeadingContainer } from "../Heading/styles";
import { Container as TextComponent } from "../TextComponent/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: 128rem;
    margin: 0 auto;
    left: 0;
    right: 0;
    & > ${TextComponent} {
      margin-bottom: ${theme.spacings.xHuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
    max-width: 120rem;
    margin: 0 auto;

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
    width: 100%;
    height: 280px;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    height: 100%;
    width: 100%;
    object-fit: cover;

    &:hover {
      transform: scale(1.2) rotate(10deg);
      transition: transform 0.5s;
    }
  `}
`;
