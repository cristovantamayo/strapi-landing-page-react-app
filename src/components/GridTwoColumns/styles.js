import styled, { css } from "styled-components";
import { Title } from "../Heading/styles";

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    justify-content: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.lessMedium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lessMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    max-width: 80rem;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css``}
`;
