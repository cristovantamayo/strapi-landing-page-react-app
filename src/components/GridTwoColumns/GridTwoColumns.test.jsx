import { screen } from "@testing-library/react";
import { describe, expect } from "@jest/globals";
import { renderTheme } from "../../styles/render-theme";
import { GridTwoColumns } from ".";
import mock from "./mock";

describe("<GridTwoColumns />", () => {
  it("should render Two columns grid", () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
