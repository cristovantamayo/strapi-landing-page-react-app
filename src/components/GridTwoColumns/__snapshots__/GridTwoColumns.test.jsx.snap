// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render Two columns grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  justify-content: center;
  gap: 3.2rem;
  max-width: 160rem;
  margin: 0 auto;
}

.c2 .c4 {
  margin-bottom: 3.2rem;
}

.c7 {
  max-width: 80rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width: 768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width: 768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width: 768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Columns
          </h2>
          <div
            class="c6"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit
          </div>
        </div>
        <div
          class="c7"
        >
          <img
            alt="Grid Two Columns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
